apps/ftu/js/wifi.js

```diff
@@ -193,35 +193,47 @@ var WifiUI = {
     window.history.back();
   },
 
-  connect: function wui_connect(ssid, password, user) {
-    // First we check if there is a previous selected network
-    // and we remove their status
-    var networkSelected = document.querySelector('li[data-wifi-selected]');
-    if (networkSelected) {
-      var icon = networkSelected.querySelector('aside');
-      networkSelected.removeAttribute('data-wifi-selected');
-      networkSelected.classList.remove('connected');
+  _selectNetworkItem: function(ssid, isSelected=true) {
+    var elem = document.getElementById(ssid);
+    if (!elem) {
+      return;
+    }
+    if (isSelected) {
+      // We should update the state to 'connecting' in here to show the change
+      // to the user, instead of waiting for the api (takes longer).
+      // But currently
+      // there's no feedback from the api when connecting to a hidden wifi fails
+      // so we skip the visual changes
+      // see https://bugzilla.mozilla.org/show_bug.cgi?id=1107431#c25
+      elem.setAttribute('data-wifi-selected', true);
+    } else {
+      var icon = elem.querySelector('aside');
+      elem.removeAttribute('data-wifi-selected');
+      elem.classList.remove('connected');
       icon.classList.remove('connecting');
       icon.classList.remove('connected');
 
-      var security = networkSelected.dataset.security;
+      var security = elem.dataset.security;
       var securityLevelDOM =
-        networkSelected.querySelectorAll('p[data-security-level]')[0];
+        elem.querySelectorAll('p[data-security-level]')[0];
```

Comment:
We could use `querySelector` to get the first element, but again it's not important, and this could add some semantics that we have more than one element that matches the selector.